account_payment_batch_process/wizard/account_payment_register.py

```diff
@@ -420,6 +420,28 @@ def make_payments(self):
                 (payment_lines + lines).filtered_domain(
                     [("account_id", "=", account.id), ("reconciled", "=", False)]
                 ).reconcile()
+            if any(
+                group_data[partner]["inv_val"][inv.id]["payment_difference_handling"]
+                == "reconcile"
+                for inv in invoices
+            ):
+                for inv in invoices:
+                    if (
+                        group_data[partner]["inv_val"][inv.id][
+                            "payment_difference_handling"
+                        ]
+                        == "reconcile"
+                    ):
```

Comment:
Here is how you can avoid the duplicate logic and still not have unnecessary loops.
```suggestion
            invoices_diff_reconcile = invoices.filtered(
                group_data[partner]["inv_val"][inv.id]["payment_difference_handling"]
                == "reconcile"
                for inv in invoices
            )
            for inv in invoices_diff_reconcile:
```